packages/electron-chrome-extensions/src/browser/router.ts

```diff
@@ -45,6 +45,12 @@ interface RoutingDelegateObserver {
   removeListener(listener: EventListener, extensionId: string, eventName: string): void
 }
 
+type PartitionSessionGrabber = (partition: string) => Electron.Session
+let getSessionFromPartition: PartitionSessionGrabber = session.fromPartition
+export function setPartitionSessionGrabber(handler: PartitionSessionGrabber) {
```

Comment:
Let's rename this to `setSessionPartitionResolver`. Also, including a JSDoc code comment for the function would help folks understand why this might be needed.